mlir/lib/Dialect/Transform/IR/TransformInterfaces.cpp

```diff
@@ -1396,14 +1396,13 @@ void transform::TrackingListener::notifyOperationReplaced(
   };
 
   // Helper function to check if the handle is alive.
-  auto hasAliveUser = [&]() {
-    for (Value v : opHandles) {
-      for (Operation *user : v.getUsers())
-        if (user != transformOp && !happensBefore(user, transformOp))
-          return true;
-    }
-    return false;
-  };
+  SmallVector<Operation *> aliveUsers;
```

Comment:
In theory, we can have a lot of users and reporting all of them isn't necessarily a readability improvement as even a dozen of notes will fill up the terminal window. How about reporting the _first found_ user instead?

Reply:
Yes I think the important piece of information here is that the is at least one user, i.e., the handle is not dead. It is then pretty easy to find that user when looking at the transform script.